fastlane/Fastfile

```diff
@@ -283,6 +286,42 @@ platform :ios do
     trigger_release_build(branch_to_build: "release/#{version}")
   end
 
+  #####################################################################################
+  # update_metadata_on_app_store_connect
+  # -----------------------------------------------------------------------------------
+  # Uploads the localized metadata tracked in the project repository to App Store
+  # Connect, optionally updating screenshots, too.
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane update_metadata_on_app_store_connect [with_screenshots:<with screenshots>]
+  #
+  # Example:
+  # bundle exec fastlane update_metadata_on_app_store_connect
+  # bundle exec fastlane update_metadata_on_app_store_connect with_screenshots:true
+  #####################################################################################
+  desc 'Updates the App Store Connect localized metadata'
+  lane :update_metadata_on_app_store_connect do |options|
+    # Skip screenshots by default. The naming is "with" to make it clear that
+    # callers need to opt-in to adding screenshots. The naming of the deliver
+    # parameter, on the other hand, uses the skip verb.
+    with_screenshots = options.fetch(:with_screenshots, false)
+    skip_screenshots = with_screenshots == false
+
+    # WordPress
+    upload_to_app_store(
+      # The reason we have these constant settings is to be able to able to
+      # share them with the Jetpack implementation (see below).
+      UPLOAD_TO_APP_STORE_SETTINGS.merge(
+        app_identifier: APP_STORE_VERSION_BUNDLE_IDENTIFIER,
+        screenshots_path: File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'promo-screenshots'),
+        skip_screenshots: skip_screenshots
+      )
+    )
+
+    # Jetpack: Delegate to dedicated lane from Jetpack-Fastfile
+    update_jetpack_metadata_on_app_store_connect(options)
```

Comment:
This is triggering my OCD a but, because we have a common lane (which acts on both WP and JP) in the common `Fastfile`, but JP-specific things separate. Why should Jetpack be separate (good idea)… but not WP (which lives in the main Fastfile which is used both for WP-only and for common lanes)…

My suggestion/idea:
 - Move the main body of this lane except those 2 jetpack lines into a `WordPressFastfile` file, into a lane called `update_wordpress_metadata_on_app_store_connect` – to mirror how you did it to JP for WP 
 - Then make this lane in the main `Fastfile` just call `update_wordpress_metadata_on_app_store_connect` then `update_jetpack_metadata_on_app_store_connect(options)`

Effectively making the `Fastfile` be the entry point that only contains lanes that act on _both_ apps at once, and keeping the code specific to WP in `WordPressFastfile` similar to how you kept the code specific to JP in `JetpackFastfile`

Or am I being too nit-picky and OCD'd here?

Reply:
I think your suggestion makes total sense @AliSoftware, it's not over the top at all.

I, too, felt it was a bit dissonant to have a mix of direct `deliver` call and delegation to an app specific lane.

The point I was coming from was that the Jetpack logic should be extracted out of the repo ASAP, so it made sense to me to leave `Fastfile` focused on WordPress and calling out to `JetpackFastfile` for the Jetpack-specific logic.

I'm probably "deluding" myself on the fact that we'll move Jetpack to its own repo soon, so it does make sense to spend a bit of time in separating the automation code. That could also be a good occasion to identify further opportunities for abstracting logic to the release toolkit level.

Reply:
Over-achiever idea:
 - Agree on a convention that every WP/JP-specific lane, which would go in their respective `*Fastfile`, follow the naming convention of starting with `wordpress_`/`_jetpack` respectively
- Then in the main `Fastfile`, use Ruby magic to generate the top level lanes which would call both WP & JP specific corresponding lane in turn) in a DRY way

I'm thinking something like:

```
# not sure about the list here, didn't check the exact cases, it's just to give the gist of it
lanes_acting_on_both_wp_jp = %i[
  upload_to_playstore
  download_metadata
  build_and_upload_to_app_store
  …
]
# auto-declare lanes dynamically, making them call both wp and jp lanes in turn
lanes_acting_on_both_wp_jp.each do |lane_name|
  desc "Call wordpress_#{lane_name} then jetpack_#{lane_name}"
  lane lane_name do |option|
    # I don't actually know the syntax to call another name given its name as a string/symbol, so strawman syntax here,
    # we'll need to search the fastlane doc on how to do this but I'm sure there's a way
  
  Shared::Lanes["wordpress_#{lane_name}"].call
    Shared::Lanes["jetpack_#{lane_name}"]
  end
end
```
(Concept code, not tested)

Again, this is me being over-the-top and all, not necessary to go that far, but just thought it was an interesting idea/implementation possibility to think about 🙃

Reply:
Actually just `send` (which is a ruby core method to call other methods using a Symbol, a bit like good ol' `[NSObject performSelector:] 😄 ) seems to be enough and work:

Tested with some dummy lanes:
```ruby
  # The kind of thing we'd put in Fastfile
  %i[foo bar].each do |lane_name|
    lane lane_name do |options|
      send("wp_#{lane_name}".to_sym, options)
      send("jp_#{lane_name}".to_sym, options)
    end
  end

  # The kind of lane we'd put in WordpressFastfile
  lane :wp_foo do |options|
    UI.message("wp_foo #{options.inspect}")
  end

  lane :wp_bar do |options|
    UI.message("wp_bar #{options.inspect}")
  end

  # The kind of lane we'd put in JetpackFastfile
  lane :jp_foo do |options|
    UI.message("jp_foo #{options.inspect}")
  end

  lane :jp_bar do |options|
    UI.message("jp_bar #{options.inspect}")
  end
```
Then testing with `fl foo x:1 y:hey`:
```
[11:51:38]: -----------------------------------
[11:51:38]: --- Step: Switch to wp_foo lane ---
[11:51:38]: -----------------------------------
[11:51:38]: Cruising over to lane 'wp_foo' 🚖
[11:51:38]: wp_foo {:x=>"1", :y=>"hey"}
[11:51:38]: Cruising back to lane 'foo' 🚘
[11:51:38]: -----------------------------------
[11:51:38]: --- Step: Switch to jp_foo lane ---
[11:51:38]: -----------------------------------
[11:51:38]: Cruising over to lane 'jp_foo' 🚖
[11:51:38]: jp_foo {:x=>"1", :y=>"hey"}
[11:51:38]: Cruising back to lane 'foo' 🚘
```